Common++/header/PointerVector.h

```diff
@@ -216,6 +301,57 @@ namespace pcpp
 		}
 
 	private:
+		/**
+		 * Performs a copy of the vector along with its elements.
+		 * The caller is responsible of freeing the copied elements.
+		 * @return A vector of pointers to the newly copied elements.
+		 */
+		static std::vector<T*> deepCopyUnsafe(std::vector<T*> const& origin)
+		{
+			std::vector<T*> copyVec;
+
+			try
+			{
+				for (const auto iter : origin)
+				{
+					T* objCopy = new T(*iter);
+					try
+					{
+						copyVec.push_back(objCopy);
+					}
+					catch (const std::exception&)
+					{
+						delete objCopy;
+						throw;
+					}
+				}
+			}
+			catch (const std::exception&)
+			{
+				for (auto obj : copyVec)
+				{
+					delete obj;
+				}
+				throw;
+			}
+
+			return copyVec;
+		}
+
+		/**
+		 * Frees all elements inside the vector.
+		 * Calling this function with non-heap allocated pointers is UB.
+		 * @param[in] origin The vector of elements to free.
+		 * @remarks The vector's contents are not cleared and will point to invalid locations in memory.
+		 */
+		static void freeVectorUnsafe(std::vector<T*> const& origin)
+		{
+			for (auto& obj : origin)
+			{
+				delete obj;
```

Comment:
do we need to check nulltpr?

Reply:
The checks happen on `pushBack`. The only other way to somehow introduce `nullptr` to the vector is via assignment to the non-`const` iterators, and if you get to that point, you are kinda deliberately shooting yourself in the foot, IMO.

Reply:
yeah.. you're right, I think my foot is blooding now XD